caikit/runtime/server_base.py

```diff
@@ -27,12 +27,38 @@
 
 # Local
 from caikit.config import get_config
+from caikit.core.exceptions import error_handler
 from caikit.runtime.model_management.model_manager import ModelManager
 from caikit.runtime.service_factory import ServicePackage, ServicePackageFactory
 from caikit.runtime.types.caikit_runtime_exception import CaikitRuntimeException
 import caikit
 
 log = alog.use_channel("SERVR-BASE")
+error = error_handler.get(log)
+
+
+class ServerThreadPool:
+    """Simple wrapper for all servers to share a single thread pool"""
+
+    @staticmethod
+    def _build_pool() -> ThreadPoolExecutor:
+        config = caikit.get_config()
+        # Leave in backwards compatibility for the old runtime.grpc.server_thread_pool_size
+        # parameter, which many users may have deployed with.
+        if pool_size := config.runtime.grpc.get("server_thread_pool_size", None):
```

Comment:
Two NITs:

1. `config.runtime.grpc.server_thread_pool_size` is exactly the same as `config.runtime.grpc.get("server_thread_pool_size", None)` because `aconfig` defines `getattr` as `get(key, None)`
2. Since this field is removed from `config.yml`, the most likely way that this backwards-compatible flag would be set (via an env var) won't work. The env vars are only read for keys that exist in the config looking for overrides.

Reply:
(Been doing too much defensive coding with @mynhardtburger these days!)